### ISSUE TITLE
Loading audio lags when using "Hide the video" button

### ISSUE THREAD

**Issue by duttyend** [Thunderarea/music-mode-for-youtube#3]
Dear dev,

First of all, thank you so much for this awesome add-on!!! It combines everything I need (skip ads + music mode), and even has a beautiful and clean interface...

I have it _**ON**_ all the time for skipping ads but I only toggle _**Hide the video**_ on some videos when not watching them.
Every time it switches from video to audio, there is a long lag. Sometimes it even stalls and it's necessary to manually click on play/pause or move the timeline to get the audio to load.

Attached you will find the settings I use.

![Screenshot 2024-01-17 115048](https://github.com/Thunderarea/music-mode-for-youtube/assets/7779942/5095a15e-a73e-4f2d-991d-39ced1d3560e)
![Screenshot 2024-01-17 115148](https://github.com/Thunderarea/music-mode-for-youtube/assets/7779942/11f0636e-9434-4db3-a1b4-51d277b85910)
![Screenshot 2024-01-17 115208](https://github.com/Thunderarea/music-mode-for-youtube/assets/7779942/6cf16022-f87a-4413-ac1b-f1f581682817)
![Screenshot 2024-01-17 115229](https://github.com/Thunderarea/music-mode-for-youtube/assets/7779942/e059e2e6-1e58-4872-ac77-ee36885cae41)
![Screenshot 2024-01-17 115247](https://github.com/Thunderarea/music-mode-for-youtube/assets/7779942/8013498b-7e60-43b1-be64-d4c956f271d1)

Thanks again for your wonderful work and happy new year!

**Comment by Thunderarea** [comment]
Thank you for your kind words. I am glad that you like it!

Regarding your issue could you please tell me which browser you are using?

In general, it needs some seconds (~1-3) to change from video to audio but not a long lag and it rarely stalls.

**Comment by duttyend** [comment]
Yes, sorry, I forgot to mention that I'm using the latest Firefox 64-bit on the latest Windows 11 version (but this has been happening since I began using your fabulous extension, so several Firefox versions have passed and nothing has changed).
I think my issue is the same as the better-phrased issue #7 you just opened.

**Comment by Thunderarea** [comment]
I have done many tests in Firefox but I never observed a so long lag (except of some cases). I will test it again. However, I know that it takes more time than in Chrome to change from video to audio. It is something that needs improvement. It is not a bug in my code but is something that depends a lot on YouTube. I have devoted many hours trying to eliminate this issue and it now works much better than in the older versions when the extension was available only for Chrome. However, there is still room for improvement. If you are a developer and you want, you are welcome to contribute/experiment towards this direction!

**Comment by duttyend** [comment]
Thanks a lot for your time and effort.
Unfortunately, I'm not a developer; hopefully, someone comes with a new idea to work around this.
Have a good evening!

**Comment by Thunderarea** [comment]
Due to changes on how YouTube works, it is not possible now to have the audio separately from the video. Therefore, this issue is not relevant anymore.
